src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -0,0 +1,288 @@
+        return (
+            this.storeForAllData &&
+            this.storeForAllData
+                .oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.isComplete &&
+            this.storeForAllData.samples.isComplete &&
+            this.storeForAllData.patients.isComplete &&
+            this.storeForAllData.coverageInformation.isComplete &&
+            this.storeForAllData.sequencedSampleKeysByGene.isComplete &&
+            this.storeForAllData.sequencedPatientKeysByGene.isComplete &&
+            this.storeForAllData.selectedMolecularProfiles.isComplete
+        );
+    }
+
+    public render() {
+        // Alteration data of non-query genes are loaded.
+        if (this.isNewStoreReady) {
+            this.addGenomicData(this.alterationFrequencyData);
+            // Toasts are removed with delay
+            setTimeout(() => {
+                this.activeToasts.forEach(tId => {
+                    toast.dismiss(tId);
+                });
+            }, 2000);
+        }
+
+        return (
+            <div className="pathwayMapper">
+                <div
+                    data-test="pathwayMapperTabDiv"
+                    className="cBioMode"
+                    style={{ width: '99%' }}
+                >
+                    <Row>
+                        <React.Fragment>
+                            <OqlStatusBanner
+                                className="coexp-oql-status-banner"
+                                store={this.props.store}
+                                tabReflectsOql={true}
+                            />
+
+                            <PathwayMapper
+                                isCBioPortal={true}
+                                isCollaborative={false}
+                                genes={this.props.store.genes.result as any}
+                                cBioAlterationData={
+                                    this.alterationFrequencyData
+                                }
+                                onAddGenes={this.handleAddGenes}
+                                changePathwayHandler={this.handlePathwayChange}
+                                addGenomicDataHandler={
+                                    this.addGenomicDataHandler
+                                }
+                                tableComponent={PathwayMapperTable}
+                                validGenes={this.validGenes}
+                                toast={toast}
+                            />
+                            <ToastContainer />
+                        </React.Fragment>
+                    </Row>
+                </div>
+            </div>
+        );
+    }
+
+    @computed get storeForAllData(): ResultsViewPageStore | undefined {
+        // Currently Pathways (PathwayMapper) tab must be active, otherwise; all toasts must be closed.
+        if (this.props.store.tabId !== ResultsViewTab.PATHWAY_MAPPER) {
+            this.activeToasts.length = 0;
+            toast.dismiss();
```

Comment:
it's really not idiomatic to make imperative changes like this in a @computed unless it's really unavoidable. can you confirm it makes logical sense to do it like this?

Reply:
will move this into a reaction defined in constructor